src/pages/resultsView/pathwayMapper/ResultsViewPathwayMapper.tsx

```diff
@@ -0,0 +1,222 @@
+        this.storeForAllData.selectedMolecularProfiles.isComplete;
+
+        if(isNewStoreReady){
+
+            this.storeForAllData.oqlFilteredCaseAggregatedDataByUnflattenedOQLLine.result!.forEach( (alterationData, trackIndex) => {
+                this.getOncoData(this.storeForAllData.samples.result,
+                                 this.storeForAllData.patients.result,
+                                 this.storeForAllData.coverageInformation.result,
+                                 this.storeForAllData.sequencedSampleKeysByGene.result!,
+                                 this.storeForAllData.sequencedPatientKeysByGene.result!,
+                                 this.storeForAllData.selectedMolecularProfiles.result!,
+                                 alterationData,
+                                 false);
+                
+            });
+
+            this.addGenomicData(this.cBioData);
+            // Toasts are removed with delay
+            setTimeout(() => {this.activeToasts.forEach(tId => {toast.dismiss(tId);});}, 1000);
+        }
+
+        // Call to this.remoteGenes.result helps remoteData to work.
+        console.log("this.remoteGenes");
+        console.log(this.remoteGenes.result);
+        return(
+
+            <div className="pathwayMapper">
+            <div data-test="pathwayMapperTabDiv" className="cBioMode" style={{width: "99%"}}>
+                <Row>
+
+                    { !this.isLoading ? 
+                    [<OqlStatusBanner style={{marginLeft: "15px"}} store={this.props.store} tabReflectsOql={true}/>,
+                        
+                     (<PathwayMapper isCBioPortal={true} isCollaborative={false} 
+                                genes={this.props.store.genes.result as any}
+                                cBioAlterationData={this.cBioData}
+                                queryParameter={QueryParameter.GENE_LIST}
+                                oncoPrintTab={ResultsViewTab.ONCOPRINT}
+                                changePathwayHandler={this.changePathwayHandler}
+                                addGenomicDataHandler={this.addGenomicDataHandler}
+                                tableComponent={PathwayMapperTable}
+                                isValidGene={this.isValidGene}
+                                toast={toast}/>),
+                        (<ToastContainer/>)
+                     ]
+                    : (<LoadingIndicator isLoading={true} size={"big"} center={true}>
+                                </LoadingIndicator>)
+                    }
+                </Row>
+            </div>
+            </div>);
+    }
+
+
+    @autobind
+    isValidGene(gene: string){
+        return this.validGenes.hasOwnProperty(gene);
+    }
+    
+    @autobind
+    addGenomicDataHandler(addGenomicData: (alterationData: ICBioData[]) => void){
+        this.addGenomicData = addGenomicData;
+    }
+
+    geneChangeHandler(genes: string[]){
```

Comment:
I don't think we need this gene change handler. You can define `storeForAllData` and `activeToasts` as `@computed` values, so that they can observe `remoteGenes.result`.